src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -866,6 +841,63 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
     }
 
+    /**
+     * Picks the cluster state with highest version from the queue. All cluster states with lower versions are ignored
```

Comment:
we need to say something about the fact that we only skip cluster states if we have a higher cluster state version from the same master